autotest/test_netcdf_gwt_henry_nr.py

```diff
@@ -0,0 +1,216 @@
+"""
+NetCDF test version of test_gwt_henry_nr.  The primary aim is to test
+that GHBA package NetCDF array input (bhead, cond, concentration and
+density auxiliary arrays) gives the same results as test_gwt_henry_nr
+list based (GHB) and array based (GHBA) ascii input runs.  This test
+compares heads in the the NetCDF file to those in the FloPy binary
+output head file.
+"""
+
+# Imports
+
+import os
+import shutil
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+try:
+    import flopy
+except:
+    msg = "Error. FloPy package is not available.\n"
+    msg += "Try installing using the following command:\n"
+    msg += " pip install flopy"
+    raise Exception(msg)
+
+from framework import TestFramework
+from test_gwt_henry_nr import cases
+
+xa = pytest.importorskip("xarray")
+xu = pytest.importorskip("xugrid")
+nc = pytest.importorskip("netCDF4")
+
+
+def build_models(idx, test, export):
+    from test_gwt_henry_nr import build_models as build
+
+    sim, mc = build(idx, test)
+    # mc.tdis.start_date_time = "2041-01-01T00:00:00-05:00"
+    gwf = mc.gwf[0]
+    gwf.get_package("GHB-1").export_array_netcdf = True
+
+    name = "gwf_" + cases[idx]
+
+    if export == "ugrid":
+        gwf.name_file.nc_mesh2d_filerecord = f"{name}.nc"
+    elif export == "structured":
+        gwf.name_file.nc_structured_filerecord = f"{name}.nc"
+
+    return sim, mc
+
+
+def check_output(idx, test, export):
+    from test_gwt_henry_nr import check_output as check
+
+    name = "gwf_" + test.name
+    ghba_ws = Path(test.workspace / "mf6")
+    ws = Path(test.workspace / "mf6" / "netcdf")
+    shutil.copytree(ghba_ws, ws)
+
+    # check outputs of GHB / GHBA ascii input runs
+    check(test.workspace, test.name, test.sims[0])
+    # check(ws, test.name, test.sims[0])
+    check(ghba_ws, test.name, test.sims[0])
+
+    # verify format of generated netcdf file
+    with nc.Dataset(ws / f"{name}.nc") as ds:
+        assert ds.data_model == "NETCDF4"
+
+    # re-run the simulation with model netcdf input
+    input_fname = f"{name}.nc"
+    nc_fname = f"{name}.{export}.nc"
+    os.rename(ws / input_fname, ws / nc_fname)
+
+    if export == "ugrid":
+        fileout_tag = "NETCDF_MESH2D"
+    elif export == "structured":
+        fileout_tag = "NETCDF_STRUCTURED"
+
+    with open(ws / f"{name}.nam", "w") as f:
```

Comment:
This seems fine for now, but getting these netcdf keywords into flopy will be nice at some point.

Reply:
Completely agree, once this PR goes in I'll revisit https://github.com/modflowpy/flopy/pull/2417, which should close that gap.